clients/src/main/java/org/apache/kafka/common/errors/IllegalClientTelemetryStateException.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+import org.apache.kafka.common.KafkaException;
+
+public class IllegalClientTelemetryStateException extends KafkaException {
```

Comment:
This exception is not mentioned on the KIP -- can it bubble up into user code -- should it be added? Or should we not even add it, and only throw an `IllegalStateException` instead?

Reply:
I don't see any exception class is mentioned in the KIP where we ll be adding some more, I am not sure if these all should be part of the KIP. However this exception should ideally not occur and specifies error in code flow itself and will only be specified in logs, not to the user application.

```
} catch (IllegalClientTelemetryStateException e) {
        log.warn("Error updating client telemetry state", e);
        return false;

Reply:
I think adding an exception should be mentioned in the kip, because the user will need to know how to handle these kind of exceptions.

A side note, we usually don't extend KafkaException.  This is to indicate the push client is in the wrong state right? `IllegalStateException` could be enough.

Reply:
I think `IllegalStateException` would be enough. It's not as if the user can do anything about this. I wouldn't add a new exception.

Reply:
I have removed the new exception class.

Reply:
Thanks! (What Philip said: we could add a new exception, but if user are supposed to handle them, it must be part of the KIP -- we could of course still add it to the KIP) -- but avoiding it and using `IllegalStateException` seems better.